index.js

```diff
@@ -43,8 +43,12 @@ HtmlWebpackPlugin.prototype.apply = function (compiler) {
   }
 
   compiler.plugin('make', function (compilation, callback) {
+    // Remove occurences of '[hash]' in the filename so the child
+    // compiler does not try to replace it.
+    var escapedFilename = self.options.filename.replace('[hash]', '');
```

Comment:
Wouldnt it be better to replace `[` and `]` to reduce the chance of collisions and support also other webpack variables?

We could move the escaping to a small file in the library folder which does only the webpack filename escaping and unescaping.

What about `[` -> `{[{` and `]` -> `}]}` ?

Should still be a valid filename: https://en.wikipedia.org/wiki/Filename